src/NzbDrone.Core/Datastore/Converters/CommandConverter.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using Marr.Data.Converters;
+using NzbDrone.Common.Reflection;
+using NzbDrone.Common.Serializer;
+using NzbDrone.Core.Messaging.Commands;
+using NzbDrone.Core.ThingiProvider;
+
+namespace NzbDrone.Core.Datastore.Converters
+{
+    public class CommandConverter : EmbeddedDocumentConverter
+    {
+        public override object FromDB(ConverterContext context)
+        {
+            if (context.DbValue == DBNull.Value)
+            {
+                return NullConfig.Instance;
+            }
+
+            var stringValue = (string)context.DbValue;
+
+            if (string.IsNullOrWhiteSpace(stringValue))
+            {
+                return NullConfig.Instance;
+            }
+
+            var ordinal = context.DataRecord.GetOrdinal("Name");
+            var contract = context.DataRecord.GetString(ordinal);
+            var impType = typeof (Command).Assembly.FindTypeByName(contract + "Command");
```

Comment:
no, leave it, from the UI perspective it's talking to a command endpoint, having a suffix there seems redundant.
Also protects against injection of stuff that isn't actually a Command.